src/System.IO.FileSystem/tests/Directory/CreateDirectory.cs

```diff
@@ -203,11 +210,29 @@ public void DirectoryWithComponentLongerThanMaxComponentAsPath_ThrowsException(s
         #region PlatformSpecific
 
         [Theory, MemberData(nameof(PathsWithInvalidColons))]
-        [PlatformSpecific(TestPlatforms.Windows)]  // invalid colons throws ArgumentException
-        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "Versions of netfx older than 4.6.2 throw an ArgumentException instead of NotSupportedException. Until all of our machines run netfx against the actual latest version, these will fail.")]
-        public void PathWithInvalidColons_ThrowsNotSupportedException(string invalidPath)
+        [PlatformSpecific(TestPlatforms.Windows)]
+        [SkipOnTargetFramework(~TargetFrameworkMonikers.NetFramework)]
+        public void PathWithInvalidColons_ThrowsNotSupportedException_Desktop(string invalidPath)
         {
-            Assert.Throws<NotSupportedException>(() => Create(invalidPath));
+            if (PathFeatures.IsUsingLegacyPathNormalization())
+            {
+                Assert.Throws<ArgumentException>(() => Create(invalidPath));
```

Comment:
Same question for all of these arg exceptions.

Reply:
For desktop the answer is no for ArgumentException. IOException doesn't have a parameter and I'm not sure we could add one in a way that made sense.